src/crypto/sign.rs

```diff
@@ -0,0 +1,135 @@
+use crate::core::matter::tables as matter;
+use crate::error::{err, Error, Result};
+
+pub(crate) fn generate(code: &str) -> Result<Vec<u8>> {
+    match code {
+        matter::Codex::Ed25519
+        | matter::Codex::Ed25519N
+        | matter::Codex::Ed25519_Seed
+        | matter::Codex::Ed25519_Sig => ed25519::generate(),
+        matter::Codex::ECDSA_256k1
+        | matter::Codex::ECDSA_256k1N
+        | matter::Codex::ECDSA_256k1_Seed
+        | matter::Codex::ECDSA_256k1_Sig => ecdsa_256k1::generate(),
```

Comment:
I chose to catch all possible codes that could apply and trust the higher levels of code to validate we are using the right type of code.